src/index.ts

```diff
@@ -1,89 +1,148 @@
+
+const run = async (newFhirMappings?) => {
+  if (newFhirMappings) {
+    await consumer.disconnect()
+    // create new consumer so we can subscribe to a (potentially) different set of topics
+    consumer = kafka.consumer({groupId: consumerGroupId})
+    fhirMappings = newFhirMappings
+  }
 
-const consumer = kafka.consumer({ groupId: "kafka-mapper-consumer" });
-const producer = kafka.producer();
+  // Register as an OpenHIM mediator and use that for config management
+  if (shouldRegisterMediator && !newFhirMappings) {
+    console.log(
+      'Registering as an OpenHIM Mediator. The OpenHIM will be used to supply latest config not the config file, the config file will provide the default config to the OpenHIM.'
+    )
+    await promisify(registerMediator)(openhimAuth, mediatorConfig)
+    const configEmitter = activateHeartbeat(openhimAuth)
 
-const run = async () => {
-  const topics = fhirMappings.map((fhirMapping) => fhirMapping.resourceType.toLowerCase());
+    configEmitter.on('config', config => {
+      console.log('Received new mapping config, restarting consumer...')
+      // restart consumer with new config
+      run(JSON.parse(config.fhirMappings))
+    })
 
-  await consumer.connect();
-  await producer.connect();
-  await consumer.subscribe({ topics, fromBeginning: true });
+    const config = await promisify(fetchConfig)(openhimAuth)
+    // set config to that received from the OpenHIM
+    fhirMappings = JSON.parse(config.fhirMappings)
+  }
+
+  const fhirMappingValidationErrors = ValidateFhirMappingsJson(fhirMappings)
+  if (fhirMappingValidationErrors.length > 0) {
+    console.error('Invalid fhir-mapping.json file')
+    fhirMappingValidationErrors.forEach(error => console.error(error))
+    process.exit(1)
+  }
+
+  const topics = fhirMappings.map(fhirMapping =>
+    fhirMapping.resourceType.toLowerCase()
+  )
+
+  await consumer.connect()
+  await consumer.subscribe({topics, fromBeginning})
 
   await consumer.run({
-    eachMessage: async ({ topic, partition, message }) => {
-      const prefix = `${topic}[${partition} | ${message.offset}] / ${message.timestamp}`;
-      console.log(`- ${prefix} ${message.key}#${message.value}`);
+    eachMessage: async ({topic, partition, message}) => {
+      const prefix = `${topic}[${partition} | ${message.offset}] / ${message.timestamp}`
+      console.log(`- ${prefix} ${message.key}#${message.value}`)
 
-      const entry: Entry = JSON.parse(message.value?.toString() ?? "");
+      const entry: Entry = JSON.parse(message.value?.toString() ?? '')
 
-      const tableMappings: Table[] = GetTableMappings(fhirMappings, entry, plugins);
+      const tableMappings: Table[] = GetTableMappings(
+        fhirMappings,
+        entry,
+        plugins
+      )
 
       for (const tableMapping of tableMappings) {
-        const res = await selectByIdClickhouse(tableMapping);
+        const res = await selectByIdClickhouse(tableMapping)
```

Comment:
Do we need a try/catch here ?

Reply:
Anything that throws in the function will get caught on line 121 and the error will be printed to the console. Are you thinkoing of any other error handling. For me that seemed enough, as there is nothing else custom that I'd want to do.